test/models/host_test.rb

```diff
@@ -1083,6 +1083,20 @@ class HostTest < ActiveSupport::TestCase
     assert_nil h.read_attribute(:root_pass), 'should not copy root_pass to host unmodified'
   end
 
+  test "should use settings root password for windows when hostgroup has empty root password" do
+    Setting[:root_pass] = "HelloWorld"
+    g = FactoryBot.create(:hostgroup, :with_domain, :with_os, :root_pass => "")
+    h = FactoryBot.create(:host, :managed, :hostgroup => g)
+    g.operatingsystem.update_attribute(:password_hash, 'Base64-Windows')
+    g.save
+    h.root_pass = ""
+    h.save
+    assert_valid h
+    assert h.root_pass.present?
+    assert_equal h.root_pass, 'SABlAGwAbABvAFcAbwByAGwAZABBAGQAbQBpAG4AaQBzAHQAcgBhAHQAbwByAFAAYQBzAHMAdwBvAHIAZABBAGQAbQBpAG4AaQBzAHQAcgBhAHQAbwByAFAAYQBzAHMAdwBvAHIAZABBAGQAbQBpAG4AaQBzAHQAcgBhAHQAbwByAFAAYQBzAHMAdwBvAHIAZABBAGQAbQBpAG4AaQBzAHQAcgBhAHQAbwByAFAAYQBzAHMAdwBvAHIAZABBAGQAbQBpAG4AaQBzAHQAcgBhAHQAbwByAFAAYQBzAHMAdwBvAHIAZAA='
```

Comment:
Due to another bug this currently encrypts `HelloWorldAdministratorPasswordAdministratorPasswordAdministratorPassword`

Will be fixed in #9584

Reply:
Adapted due to merge of https://github.com/theforeman/foreman/pull/9584